pkg/pkg.bzl

```diff
@@ -90,12 +90,28 @@ def _pkg_tar_impl(ctx):
         "--owner=" + ctx.attr.owner,
         "--owner_name=" + ctx.attr.ownername,
     ]
+    stamp_inputs = []
     if ctx.attr.mtime != _DEFAULT_MTIME:
         if ctx.attr.portable_mtime:
             fail("You may not set both mtime and portable_mtime")
+        if ctx.attr.stamp_mtime:
+            fail("You may not set both mtime and stamp_mtime")
         args.append("--mtime=%d" % ctx.attr.mtime)
     if ctx.attr.portable_mtime:
+        if ctx.attr.stamp_mtime:
+            fail("You may not set both portable_mtime and stamp_mtime")
         args.append("--mtime=portable")
+    if ctx.attr.stamp_mtime:
+        # Only ctx.version_file for file volatile-status.txt is needed
+        # because change of key/value should not invalid existing build
+        stamp_inputs += [ctx.version_file]
+        stamp_mtime_strip = ctx.attr.stamp_mtime.strip()
+        if not stamp_mtime_strip.startswith("{") or not stamp_mtime_strip.endswith("}"):
+            fail("You set stamp_mtime, but this doesn't contain a valid key variable")
```

Comment:
This should just use BUILD_STAMP.  We don't need the flexibility to let the user set an arbitrary variable out of the volatile status file. This adds complexity that no one asked for.